dist/tools/compile_and_test_for_board/compile_and_test_for_board.py

```diff
@@ -101,15 +103,41 @@ def apps_directories(riotdir, apps_dirs=None, apps_dirs_skip=None):
     :param apps_dirs_skip: list of application directories to remove, applied
                            on the RIOT list or `apps_dirs`
     """
-    apps_dirs = apps_dirs or _riot_tracked_applications_dirs(riotdir)
-    apps_dirs_skip = apps_dirs_skip or []
+    apps_dirs = _matching_apps_directories(apps_dirs, riotdir)
+    apps_dirs_skip = _matching_apps_directories(apps_dirs_skip, riotdir,
+                                                exclude=True)
 
     # Remove applications to skip
     apps_dirs = set(apps_dirs) - set(apps_dirs_skip)
 
     return sorted(list(apps_dirs))
 
 
+def _matching_apps_directories(apps_dirs, riotdir, exclude=False):
+    """Return a list of applications matching the given pattern."""
+    ret = []
+    # No specific applications directories specified, return the default
+    # list:
+    #  - empty list if this is related to excluded applications
+    #  - 'tracked' application if this is related to included applications
+    if apps_dirs is None:
+        return ret if exclude else _riot_tracked_applications_dirs(riotdir)
+    # All applications requested
+    if '*' in apps_dirs:
```

Comment:
I do not like this pattern here.

It does not make sense as an augmentation mechanism, because if you put `['a', 'b']` you get `['a', 'b']` and if you put `['a', 'b', '*']` you get the riot directories.

I want more `applications` to be a "test this that and here" only. If you want all RIOT + something, I would prefer addressing it in the RIOT side.